src/Eval.hs

```diff
@@ -38,12 +38,12 @@ data LookupPreference
   | PreferGlobal
 
 -- Prefer dynamic bindings
-evalDynamic :: Context -> XObj -> IO (Context, Either EvalError XObj)
-evalDynamic ctx xobj = eval ctx xobj PreferDynamic
+evalDynamic :: Bool -> Context -> XObj -> IO (Context, Either EvalError XObj)
```

Comment:
I think replacing the bool with a well named type would make this new code a lot more readable.